clusterloader2/framework/util/util.go

```diff
@@ -0,0 +1,159 @@
+	}
+	return RetryWithExponentialBackOff(createFunc)
+
+}
+
+// DeleteNamespace deletes namespace with given name.
+func DeleteNamespace(c clientset.Interface, namespace string) error {
+	deleteFunc := func() (bool, error) {
+		err := c.CoreV1().Namespaces().Delete(namespace, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete namespace with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+// CreateObject creates object based on given object description.
+func CreateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	obj.SetName(name)
+	createFunc := func() (bool, error) {
+		_, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(obj)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to create object with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(createFunc)
+}
+
+// DeleteObject deletes object with given name, group, version and kind.
+func DeleteObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	deleteFunc := func() (bool, error) {
+		err := dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete object with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+// GetObject retrieves object with given name, group, version and kind.
+func GetObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) (*unstructured.Unstructured, error) {
+	var obj *unstructured.Unstructured
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	getFunc := func() (bool, error) {
+		var err error
+		obj, err = dynamicClient.Resource(gvr).Namespace(namespace).Get(name, metav1.GetOptions{
+			IncludeUninitialized: true,
+		})
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to retrieve object with non-retryable error: %v", err)
+	}
+	if err := RetryWithExponentialBackOff(getFunc); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+// UpdateObject updates with given name, group, version and kind based on given object description.
+func UpdateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	// TODO(krzysied): Change it to legit update call
```

Comment:
You will never convince me that this is right and I will never let it go.

As an example: I would like to update deployment to change number of replicas from 5 to 10.
That is significantly different action that deleting that deployment (meaning removing the existing 5 pods) and then creating new one with 10.

That is just bad.

Re your comment somwhere else - yes, some update actions are not allowed (e.g. some fields are immutable). And that is fine that you can't change them.

Reply:
+1. I was pointing the exact same example in our meeting today. It seems this is "too strong" a hack to proceed on with a TODO :)

I think the best place to look for an e.g of how to update an object is kubectl (`kubectl apply -f <file>` does just this).